### PR TITLE
VM & VMSS MSI rework and support for "latest" version parameter in GetImage

### PR DESCRIPTION
**Pull request by anuchandy**
Addresses
    [1]. https://github.com/Azure/azure-sdk-for-java/issues/1844
    [2]. https://github.com/Azure/azure-sdk-for-java/issues/1392

### CODE REVIEW

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java [hunk 1]
```diff
@@ -1501,48 +1501,73 @@ WithManagedCreate withNewDataDiskFromImage(int imageLun,
...
+        /**
+         * The stage of the Managed Service Identity enabled virtual machine allowing to set role
+         * assignment for a scope.
+         */
+        @Beta(Beta.SinceVersion.V1_2_0)
+        interface WithRoleAndScopeOrCreate extends WithCreate {
             /**
-             * Specifies that Managed Service Identity needs to be enabled in the virtual machine.
+             * Specifies that applications running on the virtual machine requires the given access role
+             * with scope of access limited to the arm resource identified by the resource id specified
```
Comment (martinsawicki) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: "arm" is an acronym (ARM)

Comment (martinsawicki) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: "id" is capitalized as "ID" in user-facing text

File: azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java [hunk 2]
```diff
@@ -1501,48 +1501,73 @@ WithManagedCreate withNewDataDiskFromImage(int imageLun,
...
              * @return the next stage of the definition
              */
             @Beta(Beta.SinceVersion.V1_2_0)
-            WithCreate withManagedServiceIdentity(BuiltInRole role, String scope);
+            WithRoleAndScopeOrCreate withRoleBasedAccessToCurrentResourceGroup(BuiltInRole asRole);
 
             /**
-             * Specifies that Managed Service Identity needs to be enabled in the virtual machine.
+             * Specifies that applications running on the virtual machine requires the access described
+             * in the given role definition with scope of access limited to the arm resource identified
```
Comment (martinsawicki) [inline azure-mgmt-compute/src/main/java/com/microsoft/azure/management/compute/VirtualMachine.java]: ARM

**Comment by anuchandy** [comment]
Vlad showed that the test i skipped in fact works, propably it didn't work for me due to not copying the recording correctly. Just removed the test from ignored list and updated the recording.

**Comment by anuchandy** [comment]
@martinsawicki @milismsft @lenala @hovsepm @alvadb @jianghaolu @vladimir-shcherbakov @RikkiGibson  FYI if you have a scenario where you wanted to create an observable from a map and eventually flatmap it for parallelization, make sure the concrete type of map is TreeMap or LinkedHashMap (i.e. the DS that honor ordering). Don't use HashMap (which don't have ordering). The reason is - in playback mode, the request-response pairs are evaluated in order, which is by design.

**Comment by martinsawicki** [comment]
@anuchandy quick question -- there are no compat breaks with any previous GA or Beta functionality, right?

**Comment by anuchandy** [comment]
@martinsawicki  no, addition of MSI is not a breaking change.

**Pull request merged**